elapsed_time_block.py

```diff
@@ -71,121 +73,53 @@ def _get_timedelta(self, signal):
-                        delta['seconds'] = seconds % (delta['hours'] * 60**2)
-                    else:
-                        # seconds and hours and days
-                        delta['days'] = int(days)
-                        delta['hours'] = int(hours % (delta['days'] * 24))
-                        delta['seconds'] = seconds % (delta['hours'] * 60**2)
-                elif self.units().days(signal):
-                    # seconds and days
-                    delta['days'] = int(days)
-                    delta['seconds'] = seconds % (delta['days'] * 60**2 * 24)
-            elif self.units().minutes(signal):
-                if not self._more_significant_selected('minutes', signal):
-                    # minutes only
-                    delta['minutes'] = minutes
-                elif self.units().hours(signal):
-                    if not self._more_significant_selected('hours', signal):
-                        # minutes and hours
-                        delta['hours'] = int(hours)
-                        delta['minutes'] = minutes % (delta['hours'] * 60)
-                    else:
-                        # minutes and hours and days
-                        delta['days'] = int(days)
-                        delta['hours'] = int(hours % (delta['days'] * 24))
-                        delta['minutes'] = minutes % (delta['hours'] * 60)
-                elif self.units().days(signal):
-                    # minutes and days
-                    delta['days'] = int(days)
-                    delta['minutes'] = minutes % (delta['days'] * 60 * 24)
-            elif self.units().hours(signal):
-                if not self._more_significant_selected('hours', signal):
-                    # hours only
-                    delta['hours'] = hours
-                else:
-                    # hours and days
-                    delta['days'] = int(days)
-                    delta['hours'] = hours % (delta['days'] * 24)
-            elif self.units().days(signal):
-                # days only
-                delta['days'] = days
-        return delta
+        return seconds
+
+    def _format_seconds_diff(self, seconds, signal):
+        """ Returns the number of seconds in terms of `units` using `signal`"""
+        days_enabled = self.units().days(signal)
+        hours_enabled = self.units().hours(signal)
+        mins_enabled = self.units().minutes(signal)
+        secs_enabled = self.units().seconds(signal)
+        least_significant = "seconds"
+        least_significant_mult = 1
+
+        if not any([days_enabled, hours_enabled, mins_enabled, secs_enabled]):
+            return {}
+
+        output = {}
+
+        # Go through and remove the most significant values first. Stick the
+        # remainder on the least significant enabled value
+        if days_enabled:
+            least_significant = "days"
+            least_significant_mult = 60 * 60 * 24
+            output["days"] = days = math.floor(seconds / 60 / 60 / 24)
```

Comment:
I'm not sure about the use of `math.floor` over `int`, in the case of a negative timedelta `floor` will round down, rather than truncating. It occurs to me the tests do not cover this, so this is good behavior to define.

If the total time_delta is `-1.1` seconds, that's `-1.1 / 60` minutes:
```
>>> math.floor(-1.1 / 60)
-1
>>> int(-1.1 / 60)
0
```
So in the example that all units are selected, with a total time delta of `-1.1` seconds, the expected output would be:
```
{
  "days": 0,
  "hours": 0,
  "minutes": 0,
  "seconds": -1.1
```

It's worth pointing out that this is distinct from python's datetime.timedelta when dealing with negative intervals, the behavior of which the python docs describe as "surprising": https://docs.python.org/3/library/datetime.html#timedelta-objects

Reply:
That's a good point, I hadn't considered negatives. I wonder whether negative values are even a valid output from this block. In other words, should we just `math.abs(total_seconds)` at the beginning of the block? I could argue it either way I think.

Reply:
Yea I have other questions about how to handle negative values, come to think of it, so that'll be in another PR as well. Thanks!